03-transactions.md

```diff
@@ -783,7 +783,7 @@ The *expected weight* of a commitment transaction is calculated as follows:
 	 commitment_transaction (option_anchor_outputs): 223 + 43 * num-htlc-outputs bytes
 		- version: 4 bytes
 		- witness_header <---- part of the witness data
-		- count_tx_in: 1 byte
+		- count_tx_in: 3 byte
```

Comment:
Uh... what?  How do we get a commitment transaction with more than one input?

Did you mean to adjust count_tx_out instead?  That's definitely possible, but:

1. Implementations already have to set feerate > 250 because we calc it differently than bitcoin core brokenness ( we should add this to the spec!).  c-lightning uses 253 as the floor.  Once txs are large enough that more than compensates.
2. I think every implementation grinds sigs so they're never 73 bytes, so we are overestimating in practice anyway.

Reply:
Eh yes, `count_tx_out`, fixed.

Even though we can't cross the minimum fee rate, I think it still makes sense to fix this now that we have the opportunity to do it at virtually no cost. Alternatively we would need to document this carefully, to prevent future changes from forgetting about this hyper-optimization.